phantom/supertype.py

```diff
@@ -60,6 +63,10 @@ def to_obs_space(self, low=-np.inf, high=np.inf) -> gym.Space:
             }
         )
 
+    @property
```

Comment:
I believe you want this https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict

Reply:
`asdict` performs a deep copy on the values belonging to the dataclass so it will create copies of any samplers used.